resource/generation/client.go

```diff
@@ -600,8 +407,8 @@ func formatInterfaceTypes(types []string) string {
 
 func formatResourceInterfaceTypes(resources []resourceInfo) string {
 	names := make([]string, len(resources))
-	for i, resource := range resources {
-		names[i] = resource.Name()
+	for i := range resources {
+		names[i] = resources[i].Name()
 	}
```

Comment:
This is a bit of an anti-pattern. We should use append unless there is real utility in using an index.


```suggestion
	names := make([]string, 0, len(resources))
	for _, resource := range resources {
		names = append(names, resource.Name())
	}
```